src/dev/src/component-list.ts

```diff
@@ -0,0 +1,154 @@
+    const firstChar = component.label.charAt(0).toLowerCase();
+    const existingGroup = groups.find(g => g.label.charAt(0).toLowerCase() === firstChar);
+    if (existingGroup) {
+      existingGroup.components.push(component);
+      return groups;
+    }
+    groups.push({ label: firstChar.toUpperCase(), components: [component] });
+    return groups;
+  }, [] as IComponentGroup[]);
+}
+
+/**
+ * Filter the component groups by the provided filter text, and re-render the list.
+ */
+function onSearch(filter: string): void {
+  filter = filter.trim().toLowerCase();
+  let groups = window.structuredClone<IComponentGroup[]>(componentGroups);
+  groups = groups.filter(group => {
+    group.components = group.components.filter(({ label, tags }) => {
+      return label.toLowerCase().includes(filter) ||
+             tags?.some(tag => tag.toLowerCase().includes(filter));
+    });
+    return !!group.components.length;
+  });
+  const elements = buildComponentsList(groups);
+  listElement.replaceChildren(...elements);
+  updateContentTitle(groups, getComponentCount(groups) !== originalComponentCount);
+  updateQueryString(filter);
+}
+
+function updateQueryString(filter: string): void {
+  if (filter) {
+    url.searchParams.set('q', filter);
+  } else {
+    url.searchParams.delete('q');
+  }
+  history.pushState(null, '', url);
+}
+
+/**
+ * Updates the card header text with the component count.
+ */
+function updateContentTitle(groups: IComponentGroup[], filtered = false): number {
+  const count = getComponentCount(groups);
+  const titleText = `Components (${count}${filtered ? ` of ${originalComponentCount}` : ''})`;
+  if (filtered) {
+    contentTitleElement.innerHTML = `
+      <div class="flex">
+        ${titleText}
+        <forge-badge>Filtered</forge-badge>
+      </div>
+    `;
+  } else {
+    contentTitleElement.textContent = titleText;
+  }
+  return count;
+}
+
+/**
+ * Get a count of components across all groups.
+ */
+function getComponentCount(groups: IComponentGroup[]): number {
+  return groups.reduce((count, { components }) => count += components.length, 0);
+}
+
+/**
+ * Render the component groups.
+ */
+function buildComponentsList(groups: IComponentGroup[]): HTMLElement[] {
+  if (!groups.length) {
+    const emptyText = document.createElement('div');
+    emptyText.classList.add('forge-typography--caption');
+    emptyText.textContent = 'No components found.';
+    return [emptyText];
+  }
+
+  const elements = [];
+
+  for (const { label: groupLabel, components } of groups) {
+    const groupHeader = document.createElement('h3');
+    groupHeader.classList.add('forge-typography--subtitle1-secondary');
+    groupHeader.textContent = groupLabel;
+    elements.push(groupHeader);
+
+    for (const { label: componentLabel, path } of components) {
+      const anchor = document.createElement('a');
+      anchor.classList.add('list-item-link');
+      anchor.href = path;
+      anchor.target = '_blank';
```

Comment:
I like the new site, but tbh this `target` kind of drives me crazy.  I'd much rather use the back button than keep opening new tabs, especially when I could already ctrl/cmd click, middle-click, right-click, etc. if I did want a new tab.  And I feel like often the use case will be to test a change to a single component anyway.  But if it's not bothering anyone else, I'll survive, lol.

It also doesn't seem to work inside the embedded CodeFlow simple browser, clicking the links doesn't appear to do anything at all, though if I remove the `_blank` it does load.